### DISCUSSION TITLE
--no-git-tag-version --allow-same-version' after upgrading from version 20.1.3

### DISCUSSION THREAD

**Discussion by DanSallau** [semantic-release/semantic-release#3189] [Configuration]
Hi ,

I have `semantic-release` in our monerepo that was working find. We recenty upgraded to version `22.0.8` from version version 20.1.3 and since then our packages were not able to release. throwing the errors below

> 
'Command failed with exit code 1: npm version 2.2.0 --userconfig /tmp/3b47c665b78a38995608feea7e3b8449/.npmrc --no-git-tag-version --allow-same-version',

I have few packages in my monorepo and each package has a version ` "version": "0.0.0-development",` . 

**packages**

- widgets
- builder (has `peerDependency` widgets).

I noticed the message below from the logs

> npm WARN ERESOLVE overriding peer dependency
> npm WARN While resolving: repo/widgets@0.0.0-development
> npm WARN Found: repo/builder@undefined
> npm WARN node_modules/repo/builder
> npm WARN   workspace packages/builder from the root project
> npm WARN 

We are not updating the version in `package.json` we simply use ` "version": "0.0.0-development",`  and allow semantic-release to handle the update in the released version. I spent the whole day trying to make this upgrade works t no aveil. Any help would be appreciated

**Comment by travi** [comment]
we do not officially support monorepos. in order for us to investigate, we need you to eliminate the monorepo from the equation. setting up a public minimal reproduction would be your best bet if you want us to actually take a look. in addition, you did not upgrade to our latest major version. we do not normally backport fixes to previous major versions, so it would be best to start from our current version

**Reply by DanSallau** [reply]
@travi thanks for the prompt reply. Actually started with the latest version and got the same error hence the reason I downgraded down to other versions to in my effort to find the issue. I am wondering is giving version name as 

`"version": "0.0.0-development"` still supported in the latest version ? I don't have a semantic release config file and therefore only the `default` plugins are loaded. It seems the `"version": "0.0.0-development"` is not been resolved in my builds and is wondering if there is a change on how to do the versioning .

**Reply by travi** [reply]
> `"version": "0.0.0-development"` still supported in the latest version ? I don't have a semantic release config file and therefore only the `default` plugins are loaded. It seems the `"version": "0.0.0-development"` is not been resolved in my builds and is wondering if there is a change on how to do the versioning .

the version property in the package.json(s) in your repository is ignored by semantic-release and overwritten at publish time. it doesnt matter what value you have there as far as semantic-release is concerned. semantic-release cares about the tag that is added to your repo based on the previous release. if you are using semantic-release in a monorepo, that process is likely modified by something else in your release process beyond what semantic-release provides by default.

since this is complexity that is added beyond what this project provides, we are unwilling to support the extra complexity that you have chosen to add to your project. if you would like us to help further, you need to eliminate the monorepo from the equation to determine if the problem you are seeing is related to semantic-release or how you've approached using it with a monorepo. considering that you are seeing this problem with a previous major version and we have not seen this error reported before, it seems unlikely to be a widespread problem with supported use cases.
